src/main/java/io/spring/githubchangeloggenerator/ChangelogGenerator.java

```diff
@@ -159,8 +164,26 @@ private String getLinkToIssue(Issue issue) {
 		if (this.excludeContributors.contains("*")) {
 			return Collections.emptySet();
 		}
-		return issues.stream().filter((issue) -> issue.getPullRequest() != null).map(Issue::getUser)
-				.filter(this::isIncludedContributor).collect(Collectors.toSet());
+		return issues.stream().map(this::getPortedReferenceIssue).filter((issue) -> issue.getPullRequest() != null)
+				.map(Issue::getUser).filter(this::isIncludedContributor).collect(Collectors.toSet());
+	}
+
+	private Issue getPortedReferenceIssue(Issue issue) {
+		for (PortedIssue portedIssue : this.portedIssues) {
+			List<String> labelNames = issue.getLabels().stream().map(Label::getName).collect(Collectors.toList());
+			if (labelNames.contains(portedIssue.getLabel())) {
+				Pattern pattern = portedIssue.getBodyExpression();
+				Matcher matcher = pattern.matcher(issue.getBody());
+				if (matcher.matches()) {
+					String issueNumber = matcher.group(1);
+					Issue referencedIssue = this.service.getIssue(issueNumber, this.repository);
+					if (referencedIssue != null) {
+						return getPortedReferenceIssue(referencedIssue);
```

Comment:
@scottfrederick I don't think this would work for an issue that is a forward-port of a forward-port, would it? For example, if the original issue is in `2.2.x` and we're generating release notes for `2.4.x`.

Reply:
The recursion will follow a chain of issues that have the same label (e.g. `status: forward-port`). When the first issue in the chain *without* the label is encountered, it will be returned up the recursion stack. This does depend on the body regular expression being general enough to capture every version (e.g. `Forward port of issue #(\d+).*`) The unit test covers this case, but after another look it might be better to split out a separate test case that makes it more clear.

Reply:
oh yes, you're right. I misread what the loop does.